### PR TITLE
Adds a backfill and parameters Description annotation for the misk client

### PR DESCRIPTION
**Pull request by mpawliszyn**

### CODE REVIEW

File: client-misk/src/main/kotlin/app/cash/backfila/client/misk/Backfill.kt [hunk 1]
```diff
@@ -85,3 +85,10 @@ abstract class Backfill<E : DbEntity<E>, Pkey : Any, Param : Any> {
  * Placeholder class to represent when a backfill has no Parameters.
  */
 class NoParameters private fun constructor() { /* Never create this */ }
+
+/**
+ * Annotation used on your backfill class and parameters class constructor to send metadata to Backfila.
+ */
+@Target(AnnotationTarget.VALUE_PARAMETER, AnnotationTarget.CLASS)
```
Comment (shellderp) [inline client-misk/src/main/kotlin/app/cash/backfila/client/misk/Backfill.kt:92]: how about a file for this

Comment (mpawliszyn) [reply client-misk/src/main/kotlin/app/cash/backfila/client/misk/Backfill.kt:92]: I thought that having it part of the Backfill.kt file made sense. I can make it a separate file though.

Comment (shellderp) [reply client-misk/src/main/kotlin/app/cash/backfila/client/misk/Backfill.kt:92]: Starting to feel cluttered I guess

Comment (mpawliszyn) [reply client-misk/src/main/kotlin/app/cash/backfila/client/misk/Backfill.kt]: You like this better?

Comment (shellderp) [reply client-misk/src/main/kotlin/app/cash/backfila/client/misk/Backfill.kt]: looks good

**Pull request merged**